front/lib/api/vaults.ts

```diff
@@ -77,27 +76,75 @@ export const deleteVault = async (
       }
     }
 
+    // Soft delete data sources they will be hard deleted in the scrubbing job.
     for (const ds of dataSources) {
-      const res = await softDeleteDataSourceAndLaunchScrubWorkflow(auth, ds, t);
+      const res = await ds.delete(auth, { hardDelete: false, transaction: t });
```

Comment:
so should we use a dedicated method or delete(.., {hardDelete: false}) ?

(i rather have one way)